File: app/src/main/java/com/jason/androidstudydemos/MainActivity.java
package com.jason.androidstudydemos;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.AbsListView;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ListView listView;
    ImageView imageView;
    List<Msg> list = new ArrayList<>();
    int[] imgArr = {R.mipmap.profile1, R.mipmap.profile2,
            R.mipmap.profile3, R.mipmap.profile4,
            R.mipmap.profile5, R.mipmap.profile6,
            R.mipmap.profile7, R.mipmap.profile8};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.list_view);
        imageView = findViewById(R.id.add_img);

        initData();

        BaseAdapter adapter = new MyBaseAdapter(this, list);

        listView.setAdapter(adapter);

        imageView.setOnClickListener(v -> {
            list.add(new Msg(R.mipmap.profile9, "虎正玺", "你好我好大家好", false));
            adapter.notifyDataSetChanged();
            listView.setTranscriptMode(AbsListView.TRANSCRIPT_MODE_ALWAYS_SCROLL);
        });
    }

    private void initData() {
        for (int i = 1; i < 9; i++) {
            Msg msg = new Msg(imgArr[i - 1], "用户" + i, "说说" + i, i % 2 == 1);
            list.add(msg);
        }
    }
}

File: animationdemo/src/main/java/com/jason/animationdemo/MainActivity.kt
package com.jason.animationdemo

import android.content.Intent
import android.graphics.drawable.AnimationDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.jason.animationdemo.frame.FrameAnimationActivity
import com.jason.animationdemo.transitions.FirstActivity
import com.jason.animationdemo.view.ViewAnimationActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        id_btn_frame.setOnClickListener {
            startActivity(Intent(this, FrameAnimationActivity::class.java))
        }

        id_btn_view.setOnClickListener {
            startActivity(Intent(this, ViewAnimationActivity::class.java));
        }

        id_btn_transitions.setOnClickListener {
            startActivity(Intent(this, FirstActivity::class.java))
        }
    }
}